### PR TITLE
feat: [SystemFontProvider] - Implement GetFontPath for Linux

### PR DESCRIPTION
**Pull request by Jklawreszuk**
# PR Details

This PR refactors GetFontPath to find system fonts for Linux for Asset Compiler.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] **I have built and run the editor to try this change out.**

### CODE REVIEW

File: sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs [hunk 1]
```diff
@@ -52,72 +50,105 @@ public SystemFontProvider(string fontName)
...
-                        var weight = Style.IsBold() ? FontWeight.Bold : FontWeight.Regular;
-                        var style = Style.IsItalic() ? SharpDX.DirectWrite.FontStyle.Italic : SharpDX.DirectWrite.FontStyle.Normal;
-                        font = fontFamily.GetFirstMatchingFont(weight, FontStretch.Normal, style);
-                        if (font == null)
-                        {
-                            result?.Error($"Cannot find style '{Style}' for font family {FontName}. Make sure it is installed on this machine.");
-                            return null;
-                        }
-                    }
+                    return file;
```
Comment (Eideren) [inline sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs:101]: I'm guessing there are less hacky ways to get the style from a font, probably have to read the .ttf though ? If that's the case, can you add a todo here for this ?

Insert a line break between this function and the previous one.
`GetFiles` is fairly inefficient for this specific use case, best replace it with a `Directory.EnumerateFiles(systemFontDirectory, "*.ttf", System.IO.SearchOption.AllDirectories)` instead.

Comment (Jklawreszuk) [reply sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs]: > I'm guessing there are less hacky ways to get the style from a font, probably have to read the .ttf though

Yeah, I'll try to see if it is possible to verify the style with our current FreeType binding, but common convetion I noticed is also separating styles into multiple files. I'm not though sure is always the case - see C:\Windows\Fonts where all fonts have single file for each style.

Perhaps I should also consider adding a specific font to the Search regex. Something like this:
`Directory.EnumerateFiles(systemFontDirectory, $"{FontName}*.ttf", System.IO.SearchOption.AllDirectories)`

File: sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 // Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
-
 using System.Linq;
 using SharpDX.DirectWrite;
 using Stride.Core.Assets.Compiler;
 using Stride.Core;
 using Stride.Core.Diagnostics;
 using Stride.Assets.SpriteFont.Compiler;
 using Stride.Graphics.Font;
+using System;
```
Comment (Kryptos-FR) [inline sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs:10]: Nit: sort usings

File: sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs [hunk 3]
```diff
@@ -52,72 +51,111 @@ public SystemFontProvider(string fontName)
...
+                font = fontFamily.GetFirstMatchingFont(weight, FontStretch.Normal, style);
             }
 
             return new FontFace(font);
         }
 
         public override string GetFontPath(AssetCompilerResult result = null)
         {
-            using (var factory = new Factory())
+            if(OperatingSystem.IsWindows())
```
Comment (Kryptos-FR) [inline sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs:76]: Nit: code style, space after opening bracket

File: sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs [hunk 4]
```diff
@@ -52,72 +51,111 @@ public SystemFontProvider(string fontName)
...
-                    }
+                    result?.Warning($"Cannot find font family '{FontName}'. Loading default font '{GetDefaultFontName()}' instead");
+                    FontName = GetDefaultFontName();
+                    fontPath = GetFontPathLinux(result);
                 }
+                return fontPath;
+            }
+            return null;
+        }
+        private string GetFontPathLinux(AssetCompilerResult result)
```
Comment (Kryptos-FR) [inline sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs:91]: NIt: code style, empty line between methods

Comment (Jklawreszuk) [reply sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs:92]: Thanks, I  added new line

File: sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs (hunk 2 above)
Comment (Jklawreszuk) [reply sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs:10]: done!

File: sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs (hunk 3 above)
Comment (Jklawreszuk) [reply sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs:76]: done?

File: sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs (hunk 2 above)
Comment (Kryptos-FR) [reply sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs:10]: Usually. System usings are first and then the rest in dictionary order. But that's ok here.

We might even use implicit usings later which would hide those.

File: sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs (hunk 3 above)
Comment (Kryptos-FR) [reply sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs:76]: I meant `if (` instead of `if(`

Comment (Jklawreszuk) [reply sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs]: Right 🤦 Done, I also updated other if statements

**Comment by Eideren** [comment]
Thanks !

**Pull request merged**